autofs-5.1.8 - get rid entry thid field

From: Ian Kent <raven@themaw.net>

Use the autofs_point structure thid and get rid of the copy in struct
master_mapent.

Signed-off-by: Ian Kent <raven@themaw.net>
---
 CHANGELOG        |    1 +
 daemon/master.c  |    5 ++---
 include/master.h |    1 -
 3 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/CHANGELOG b/CHANGELOG
index ca72023b..ba6bab3b 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -74,6 +74,7 @@
 - fix memory leak in sasl_do_kinit()
 - fix fix mount tree startup reconnect.
 - fix amd selector function matching.
+- get rid entry thid field.
 
 19/10/2021 autofs-5.1.8
 - add xdr_exports().
diff --git a/daemon/master.c b/daemon/master.c
index 6885f7c7..60f69648 100644
--- a/daemon/master.c
+++ b/daemon/master.c
@@ -837,7 +837,6 @@ struct master_mapent *master_new_mapent(struct master *master, const char *path,
 	entry->path = tmp;
 	entry->len = strlen(tmp);
 
-	entry->thid = 0;
 	entry->age = age;
 	entry->master = master;
 	entry->current = NULL;
@@ -1422,7 +1421,7 @@ static int master_do_mount(struct master_mapent *entry)
 		handle_mounts_startup_cond_destroy(&suc);
 		return 0;
 	}
-	entry->thid = ap->thid = thid;
+	ap->thid = thid;
 
 	handle_mounts_startup_cond_destroy(&suc);
 
@@ -1977,7 +1976,7 @@ int master_done(struct master *master)
 		entry = list_entry(p, struct master_mapent, join);
 		p = p->next;
 		list_del(&entry->join);
-		pthread_join(entry->thid, NULL);
+		pthread_join(entry->ap->thid, NULL);
 		master_free_mapent_sources(entry, 1);
 		master_free_mapent(entry);
 	}
diff --git a/include/master.h b/include/master.h
index befeefa2..0bdef42e 100644
--- a/include/master.h
+++ b/include/master.h
@@ -46,7 +46,6 @@ struct map_source {
 struct master_mapent {
 	char *path;
 	size_t len;
-	pthread_t thid;
 	time_t age;
 	struct master *master;
 	pthread_rwlock_t source_lock;
